assets/sass/widgets/_googlesitekit-widget-analyticsAllTraffic.scss

```diff
@@ -167,7 +167,10 @@
 		position: absolute;
 		text-align: center;
 		text-transform: uppercase;
-		top: 50%;
+		// The -6px margin-top on the legend affects the parent height
+		// so we have to factor it in here to center the title
+		top: calc(50% - 6px);
```

Comment:
I'm not sure this is the right solution here. Looking at the component, the title seems to be intended to be a child of the chart component and the legend below:
https://github.com/google/site-kit-wp/blob/0075f6de4035b67d5830ada399bbc8cdca9c5a37/assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidget/UserDimensionsPieChart.js#L387-L410
However, due to an implementation detail of the chart being output as a fragment with the chart followed by the children, this results in some unexpected output where the title and legend are siblings of each other which causes the legend to influence the height of the pie chart when it really shouldn't.

I would suggest we update `GoogleChartV2` to provide a wrapping `div` instead so that the title is always relative to the area of the chart.

<details>
<summary>This worked well in my testing:</summary>

![Kapture 2021-03-03 at 13 00 25](https://user-images.githubusercontent.com/1621608/109796928-6ed2b600-7c21-11eb-8324-ab73f90ed42a.gif)

</details>
@tofumatt any reason we can't do something like this?

<details>
<summary>git diff</summary>

```diff
diff --git a/assets/js/components/GoogleChartV2.js b/assets/js/components/GoogleChartV2.js
index cfef6c26d..4e2fcc4fc 100644
--- a/assets/js/components/GoogleChartV2.js
+++ b/assets/js/components/GoogleChartV2.js
@@ -136,13 +136,19 @@ export default function GoogleChartV2( props ) {
 	}
 
 	return (
-		<Fragment>
+		<div
+			className={ classnames(
+				'googlesitekit-chart-v2',
+				`googlesitekit-chart-v2--${ chartType }`,
+				className
+			) }
+		>
 			<Chart
 				chartEvents={ combinedChartEvents }
 				chartType={ chartType }
 				className={ classnames(
-					'googlesitekit-chart-v2',
-					`googlesitekit-chart-v2--${ chartType }`,
+					'googlesitekit-chart-v2__chart',
+					`googlesitekit-chart-v2__chart--${ chartType }`,
 					className
 				) }
 				loader={ loader }
@@ -170,7 +176,7 @@ export default function GoogleChartV2( props ) {
 				{ ...otherProps }
 			/>
 			{ children }
-		</Fragment>
+		</div>
 	);
 }
 
diff --git a/assets/sass/components/global/_googlesitekit-chart.scss b/assets/sass/components/global/_googlesitekit-chart.scss
index 03294e911..7d083cf0b 100644
--- a/assets/sass/components/global/_googlesitekit-chart.scss
+++ b/assets/sass/components/global/_googlesitekit-chart.scss
@@ -24,6 +24,10 @@
 	}
 }
 
+.googlesitekit-chart-v2 {
+	position: relative;
+}
+
 .googlesitekit-chart-v2--PieChart svg > g > g {
 	pointer-events: none;
 }
```
</details>

Reply:
No reason at all that shouldn't work, though we'll want to adjust a single CSS rule we have that has Hyper-Selector-Specificity™: https://github.com/google/site-kit-wp/blob/0075f6de4035b67d5830ada399bbc8cdca9c5a37/assets/sass/widgets/_googlesitekit-widget-analyticsAllTraffic.scss#L144-L150

I think it's a matter of adding another `> div` to make up for what `Chart` renders. We have to be really specific there to override and target specific SVG styles, if I recall correctly. (Alternatively, add a `googlesitekit-chart-v2__inner` to `Chart`'s `className` or something and use that.)

(Also: I'd strip out the `className` from the `Chart` component entirely there rather than duplicate it in the `div` and `Chart` component, but I think that might've been done by mistake 😅)

TL;DR is yes replacing the `Fragment` with a `<div>` should be fine 👍🏻

Reply:
Thanks @tofumatt 
Right, let me give this a whirl...